src/ComboBox/index.jsx

```diff
@@ -77,438 +80,390 @@ function optionsFormatted( filteredOptionsIds, originalOptions )
-            activeOption    : undefined,
-            filteredOptions : undefined,
-            flatOptions     : undefined,
-            id              : undefined,
-            isOpen          : undefined,
-            options         : undefined,
-            searchValue     : undefined,
-            selectedOption  : undefined,
-        };
-
-        this.handleBlur            = this.handleBlur.bind( this );
-        this.handleChangeInput     = this.handleChangeInput.bind( this );
-        this.handleClick           = this.handleClick.bind( this );
-        this.handleClickIcon       = this.handleClickIcon.bind( this );
-        this.handleClickOption     = this.handleClickOption.bind( this );
-        this.handleKeyDown         = this.handleKeyDown.bind( this );
-        this.handleMouseOutOption  = this.handleMouseOutOption.bind( this );
-        this.handleMouseOverOption = this.handleMouseOverOption.bind( this );
-    }
-
-    static getDerivedStateFromProps( props, state )
-    {
-        let { flatOptions } = state;
-        const { selectedOption } = props;
-        let optionId = selectedOption || state.selectedOption;
-
-        if ( props.options !== state.options )
-        {
-            flatOptions = props.options.flatMap( o => o.options || o );
-        }
-
-        if ( optionId )
+        if ( !selectedOption )
         {
-            optionId = getOption( optionId, flatOptions ) ?
-                getOption( optionId, flatOptions ).id : undefined;
+            setOption( value );
         }
+    };
 
-        return {
-            flatOptions,
-            filteredOptions : state.filteredOptions,
-            id              : props.id || state.id || generateId( 'ComboBox' ),
-            options         : props.options,
-            searchValue     : state.searchValue,
-            selectedOption  : optionId,
-        };
-    }
+    return [ selectedOption || option, setter ];
+};
 
-    componentDidUpdate()
+const ComboBox = props =>
+{
+    const [ activeOption, setActiveOption ] = useState( undefined );
+    const [ filteredOptions, setFilteredOptions ] = useState( undefined );
+    const [ isOpen, setIsOpen ] = useState( undefined );
+    const [ searchValue, setSearchValue ] = useState( undefined );
+
+    const inputRef = useRef( null );
+    const scrollBoxRef = useRef( null );
+    const wrapperRef = useRef( null );
+
+    const {
+        className,
+        dropdownPlaceholder,
+        dropdownPosition,
+        hasError,
+        id,
+        inputPlaceholder,
+        isDisabled,
+        isReadOnly,
+        isSearchable,
+        onChange,
+        options,
+        selectedOption,
+    } = props;
+
+    const [ optionSelected,
+        setOptionSelected ] = useSelectedOption( selectedOption );
+
+    const flatOptions = useMemo(
+        () => ( options.flatMap( o => o.options || o ) ),
+        [ options ],
+    );
+
+    const componentId = useMemo(
+        () => ( id || generateId( 'ComoBox' ) ),
```

Comment:
C-C-C-ComboBreaker!!!!
```suggestion
        () => ( id || generateId( 'ComboBox' ) ),
```